client/modules/IDE/components/FileNode.jsx

```diff
@@ -10,6 +10,47 @@ import FolderRightIcon from '../../../images/triangle-arrow-right.svg';
 import FolderDownIcon from '../../../images/triangle-arrow-down.svg';
 import FileIcon from '../../../images/file.svg';
 
+function FileName({ name }) {
+  const nameArray = name.split('.');
+  if (nameArray.length > 1) {
+    const extension = `.${nameArray[nameArray.length - 1]}`;
```

Comment:
You can use nameArray.slice(-1) here, if that's more readable

Reply:
I always forget that I can use `slice` for this 😄

Reply:
I think I'm actually going to change this to `nameArray.charAt(nameArray.length - 1)` as I think that's the most readable.

Reply:
Wow, so apparently `charAt` can work in some really wacky ways (see this [StackOverflow answer](https://stackoverflow.com/a/30742383)), so......... I'm just going to stick with bracket notation.

Reply:
Make sure to check/test what happens if you pass an empty string to it!
Javascript doesn't like `array[-1]`

Reply:
Nevermind, you're catching this on line 15. My bad!